engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -964,7 +965,13 @@ else if (allocatedContainers.isEmpty() && numRequestedContainers == 0 && dnmgr.c
         appDone = true;
       }
 
-      LOG.debug("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" + numTotalContainers + ", requested=" + numRequestedContainers + ", released=" + numReleasedContainers + ", completed=" + numCompletedContainers + ", failed=" + numFailedContainers + ", currentAllocated=" + allocatedContainers.size());
+      if (!StringUtils.isEmpty(dnmgr.shutdownDiagnosticsMessage) && !appDone) {
```

Comment:
It is less readable in the code when there are multiple LOG statements and it should be possible to make log entry readable without multiple log statements. In this particular case, it should be sufficient to log necessary variables values, IMO.